xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -1294,6 +1294,7 @@ void CVideoPlayer::Process()
       CSingleLock lock(m_StateSection);
       if (m_displayLost)
       {
+        CSingleExit unlock(m_StateSection);
```

Comment:
You would use CSingleExit if you want to exit a lock and grab it again. This is not the case here. It would grab the lock again and on the next line the outer lock gets destructed anyway.